### PR TITLE
sensc: fixing and checking gps data

### PR DESCRIPTION
**Pull request by niewim19**
## Description

 - fixing `sensc` gps file descriptor bug
 - adding `senslog` gps logging capability (only latitude/longitude/satelites/fix are printed) 

## Motivation and Context

This PR adds testing capability of GPS device output

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: zturn drone.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
  - https://github.com/phoenix-rtos/phoenix-rtos-devices/pull/342
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: utils/senslog.c [hunk 1]
```diff
@@ -229,6 +240,11 @@ int main(int argc, char **argv)
 			cnt = (res < 0) ? -1 : cnt + res;
 		}
 
+		if (senslog_common.flags.gps && cnt >= 0) {
+			res = sprintf(&buf[cnt], "%d %d %u %u", gpsEvt.gps.lat, gpsEvt.gps.lon, gpsEvt.gps.satsNb, gpsEvt.gps.fix);
```
Comment (agkaminski) [inline utils/senslog.c:244]: This has a potential to generate up to 47 characters in total. Is `buf` still big enough? There're no checks for cnt, this is somewhat ok if we design `buf` size for a worst case scenario.

Comment (niewim19) [reply utils/senslog.c]: Thanks. That is indeed a problem. Current worst-case scenario was 182 characters printed. Added calculation comments and comment regarding adjusting the buffer if needed.

Resized buffer to 256 characters for now.

File: utils/senslog.c [hunk 2]
```diff
@@ -200,40 +207,53 @@ int main(int argc, char **argv)
...
+		}
+
+		if (senslog_common.flags.gps && cnt >= 0) {
+			res = sprintf(&buf[cnt], "%d %d %u %u ", gpsEvt.gps.lat, gpsEvt.gps.lon, gpsEvt.gps.satsNb, gpsEvt.gps.fix); /* Max 42 characters */
 			cnt = (res < 0) ? -1 : cnt + res;
 		}
 
 		if (cnt >= 0) {
-			res = sprintf(&buf[cnt], "\n");
+			res = sprintf(&buf[cnt], "\n"); /* Max 1 character */
```
Comment (agkaminski) [inline utils/senslog.c:249]: Nitpick - two characters (NULL terminator is not eaten here by next `sprintf`s)

File: utils/senslog.c [hunk 3]
```diff
@@ -200,40 +207,53 @@ int main(int argc, char **argv)
...
 		}
 
 		if (senslog_common.flags.baro && cnt >= 0) {
-			res = sprintf(&buf[cnt], "%d %d ", baroEvt.baro.pressure, baroEvt.baro.temp);
+			res = sprintf(&buf[cnt], "%d %d ", baroEvt.baro.pressure, baroEvt.baro.temp); /* Max 21 characters */
+			cnt = (res < 0) ? -1 : cnt + res;
+		}
+
+		if (senslog_common.flags.gps && cnt >= 0) {
+			res = sprintf(&buf[cnt], "%d %d %u %u ", gpsEvt.gps.lat, gpsEvt.gps.lon, gpsEvt.gps.satsNb, gpsEvt.gps.fix); /* Max 42 characters */
```
Comment (agkaminski) [inline utils/senslog.c:244]: Nitpick - 46, %d can produce max 11 chars for negative number

File: utils/senslog.c [hunk 4]
```diff
@@ -200,40 +207,53 @@ int main(int argc, char **argv)
...
 
 		if (senslog_common.flags.mag && cnt >= 0) {
-			res = sprintf(&buf[cnt], "%d %d %d ", magEvt.mag.magX, magEvt.mag.magY, magEvt.mag.magZ);
+			res = sprintf(&buf[cnt], "%d %d %d ", magEvt.mag.magX, magEvt.mag.magY, magEvt.mag.magZ); /* Max 31 characters */
 			cnt = (res < 0) ? -1 : cnt + res;
 		}
 
 		if (senslog_common.flags.baro && cnt >= 0) {
-			res = sprintf(&buf[cnt], "%d %d ", baroEvt.baro.pressure, baroEvt.baro.temp);
+			res = sprintf(&buf[cnt], "%d %d ", baroEvt.baro.pressure, baroEvt.baro.temp); /* Max 21 characters */
```
Comment (agkaminski) [inline utils/senslog.c:239]: *24 (ignoring null terminator)

File: utils/senslog.c [hunk 5]
```diff
@@ -200,40 +207,53 @@ int main(int argc, char **argv)
...
 
 		if (senslog_common.flags.gyro && cnt >= 0) {
-			res = sprintf(&buf[cnt], "%d %d %d ", gyroEvt.gyro.gyroX, gyroEvt.gyro.gyroY, gyroEvt.gyro.gyroZ);
+			res = sprintf(&buf[cnt], "%d %d %d ", gyroEvt.gyro.gyroX, gyroEvt.gyro.gyroY, gyroEvt.gyro.gyroZ); /* Max 31 characters */
 			cnt = (res < 0) ? -1 : cnt + res;
 		}
 
 		if (senslog_common.flags.mag && cnt >= 0) {
-			res = sprintf(&buf[cnt], "%d %d %d ", magEvt.mag.magX, magEvt.mag.magY, magEvt.mag.magZ);
+			res = sprintf(&buf[cnt], "%d %d %d ", magEvt.mag.magX, magEvt.mag.magY, magEvt.mag.magZ); /* Max 31 characters */
```
Comment (agkaminski) [inline utils/senslog.c:234]: *36

File: utils/senslog.c [hunk 6]
```diff
@@ -200,40 +207,53 @@ int main(int argc, char **argv)
...
 
 		if (senslog_common.flags.accel && cnt >= 0) {
-			res = sprintf(&buf[cnt], "%d %d %d ", accelEvt.accels.accelX, accelEvt.accels.accelY, accelEvt.accels.accelZ);
+			res = sprintf(&buf[cnt], "%d %d %d ", accelEvt.accels.accelX, accelEvt.accels.accelY, accelEvt.accels.accelZ); /* Max 31 characters */
 			cnt = (res < 0) ? -1 : cnt + res;
 		}
 
 		if (senslog_common.flags.gyro && cnt >= 0) {
-			res = sprintf(&buf[cnt], "%d %d %d ", gyroEvt.gyro.gyroX, gyroEvt.gyro.gyroY, gyroEvt.gyro.gyroZ);
+			res = sprintf(&buf[cnt], "%d %d %d ", gyroEvt.gyro.gyroX, gyroEvt.gyro.gyroY, gyroEvt.gyro.gyroZ); /* Max 31 characters */
```
Comment (agkaminski) [inline utils/senslog.c:229]: *36

File: utils/senslog.c [hunk 7]
```diff
@@ -200,40 +207,53 @@ int main(int argc, char **argv)
...
 		cnt = 0;
 
 		/* printing time as a fake float value */
-		res = sprintf(&buf[cnt], "%lli.%06lli ", diff / 1000000, (diff - 1000000 * (diff / 1000000)));
+		res = sprintf(&buf[cnt], "%lli.%06lli ", diff / 1000000, (diff - 1000000 * (diff / 1000000))); /* Max 26 characters */
 		cnt = (res < 0) ? -1 : cnt + res;
 
 		if (senslog_common.flags.accel && cnt >= 0) {
-			res = sprintf(&buf[cnt], "%d %d %d ", accelEvt.accels.accelX, accelEvt.accels.accelY, accelEvt.accels.accelZ);
+			res = sprintf(&buf[cnt], "%d %d %d ", accelEvt.accels.accelX, accelEvt.accels.accelY, accelEvt.accels.accelZ); /* Max 31 characters */
```
Comment (agkaminski) [inline utils/senslog.c:224]: *36

**Review by agkaminski — APPROVED** [review]

**Comment by niewim19** [comment]
Yeah, I missed the _sign_ sign in calculations for `%d`. Thanks! Added 2 characters for `%lli` as it technically can be negative.

**Pull request merged**